application/temperature.py

```diff
@@ -0,0 +1,36 @@
+import httpx
+from sqlalchemy.orm import Session
+from datetime import datetime
+from application import models, schemas
+
+
+async def fetch_temperature(city_name: str, api_key: str) -> float | None:
+    url = "http://api.openweathermap.org/data/2.5/weather"
```

Comment:
Consider using a configuration file or environment variable to manage the API URL, which will make the application more flexible and secure.